autode/opt/optimisers/dimer.py

```diff
@@ -0,0 +1,359 @@
+            raise ValueError('A dimer optimisation must be initialised from '
+                             'a set of dimer coordinates')
+
+        optimiser = cls(maxiter=100, coords=coords)
+        optimiser.run(species=species, method=method, n_cores=n_cores)
+        return None
+
+    def _step(self) -> None:
+        """
+        Do a single dimer optimisation step, consisting of several rotation and
+        translation steps.
+        """
+        self._update_gradient_at(DimerPoint.left)
+
+        self._optimise_rotation()
+        trns_result = self._translate()
+
+        if trns_result == _StepResult.skipped_translation:
+            self._converged_translation = True
+
+        return None
+
+    def _rotate(self) -> '_StepResult':
+        """Apply a rotation"""
+
+        c_phi0, dc_dphi0 = self._c, self._dc_dphi
+        logger.info('Doing a single dimer rotation to minimise the curvature. '
+                    f'Current: C = {c_phi0:.4f} '
+                    f'and dC/dϕ = {dc_dphi0:.4f}')
+
+        cached_coordinates = self._coords.copy()
+
+        phi_1 = self._phi1.to('radians')
+        logger.info(f'Rotating by ϕ = {phi_1.to("degrees"):.4f}º and '
+                    f'evaluating the curvature')
+
+        if abs(phi_1) < self.phi_tol:
+            logger.info('Rotation angle was below the threshold, not rotating')
+            return _StepResult.skipped_rotation
+
+        self._rotate_coords(phi_1, update_g1=True)
+
+        b1 = 0.5 * dc_dphi0                              # eqn. 8 from ref. [1]
+
+        a1 = ((c_phi0 - self._c + b1 * np.sin(2*phi_1))  # eqn. 9 from ref. [1]
+              / (1 - 2.0 * np.cos(2.0 * phi_1)))
+
+        a0 = 2.0 * (c_phi0 - a1)                        # eqn. 10 from ref. [1]
+        phi_min = Angle(0.5 * np.arctan(b1 / a1), units='radians')
+        logger.info(f'ϕ_min = {phi_min.to("degrees"):.4f}º')
+
+        c_min = 0.5 * a0 + a1 * np.cos(2.0*phi_min) + b1 * np.sin(2.0*phi_min)
+
+        if c_min > c_phi0:
+            logger.info('Optimised curvature was larger than the initial, '
+                        'adding π/2')
+            phi_min += np.pi / 2.0
+
+        # Rotate back from the test point, then to the minimum
+        self._coords = cached_coordinates
+        self._rotate_coords(phi=phi_min, update_g1=True)
+
+        return _StepResult.did_rotation
+
+    def _translate(self, update_g0=True) -> '_StepResult':
+        """Translate the dimer under the translational force"""
+        x0 = self._coords.x0
+
+        trns_iters = [c for c in self._history if c.did_translation]
+
+        if len(trns_iters) < 2:
+            step_size = float(self.init_alpha.to("Å"))
+
+            logger.info(f'Did not have two previous translation step, guessing'
+                        f' γ = {step_size} Å')
+
+        else:
+            prev_trns_iter = trns_iters[-2]
+            logger.info(f'Did {len(trns_iters)} previous translations, can '
```

Comment:
Maybe this is not a good idea for a dimer, where the step is not quite a steepest decent along `f_t`